src/NGen.cpp

```diff
@@ -541,11 +541,14 @@ int main(int argc, char* argv[]) {
     // T-ROUTE data storage
     std::unordered_map<std::string, int> nexus_indexes;
 #if NGEN_WITH_ROUTING
-    size_t nexus_collection_size = nexus_collection->get_size();
-    for (int i = 0; i < nexus_collection_size; ++i) {
+    int nexus_index = 0;
+    for (int i = 0; i < nexus_collection->get_size(); ++i) {
         auto feature = nexus_collection->get_feature(i);
```

Comment:
```suggestion
        auto const& feature = nexus_collection->get_feature(i);
```
No need to take a copy.